src/component/reducer/Reducer.ts

```diff
@@ -457,37 +455,20 @@ export function dispatchMiddleware(dispatch) {
 
         break;
       }
-      case types.LOAD_JSON_FILE: {
-        const parsedData = JSON.parse(action.files[0].binary.toString());
-        const data = migrate(parsedData);
-        void SpectraManager.fromJSON(data.spectra, usedColors).then(
-          (spectra) => {
-            action.payload = Object.assign(data, { spectra, usedColors });
-            dispatch(action);
-          },
-        );
-        break;
-      }
-      case types.LOAD_ZIP_FILE: {
-        for (let zipFile of action.files) {
-          void SpectraManager.addBruker(
-            { display: { name: zipFile.name } },
-            zipFile.binary,
-            usedColors,
-          ).then((data) => {
-            action.payload = { data, usedColors };
+      case types.LOAD_DROP_FILES: {
+        const { files } = action;
+        action.usedColors = usedColors;
+        try {
+          void readDropFiles(files, usedColors).then((data) => {
+            action.data = data;
             dispatch(action);
           });
+        } catch (e: any) {
```

Comment:
You cannot `try...catch` an async error unless you use `await`.

Here you can't do that because it's not an async function. So handle your error in `readDropFiles(...).catch(...)`

Please test that error handling is working properly by throwing a random error inside the `then` handler.